package app.urv.manager.util

const val tag = "Universal ReVanced Manager"

const val JAR_MIMETYPE = "application/java-archive"
const val APK_MIMETYPE = "application/vnd.android.package-archive"
// Code adapted from Morphe, see third-party/NOTICE for more information
// https://github.com/MorpheApp/morphe-manager/commit/7e24461c1454b712da4df21440db6f417c94ce58
const val PLAY_STORE_INSTALLER_PACKAGE = "com.android.vending"
const val APK_SIGNER_CACHE_DIR = "apk-signer"
const val APK_SIGNATURE_METADATA_INJECTOR_CACHE_DIR = "apk-signature-metadata-injector"
val APK_FILE_MIME_TYPES = arrayOf(
    APK_MIMETYPE,
    "application/zip",
    "application/x-zip-compressed",
    "application/x-apkm",
    "application/x-apks",
    "application/x-xapk",
    "application/xapk",
    "application/vnd.android.xapk",
    "application/vnd.android.apkm",
    "application/apkm",
    "application/vnd.android.apks",
    "application/apks",
    BIN_MIMETYPE
)
val APK_FILE_EXTENSIONS = setOf(
    "apk",
    "apkm",
    "apks",
    "xapk",
    "zip"
)
val SPLIT_ARCHIVE_FILE_EXTENSIONS = setOf(
    "apkm",
    "apks",
    "xapk",
    "zip"
)
val SPLIT_ARCHIVE_MIME_TYPES = arrayOf(
    "application/zip",
    "application/x-zip-compressed",
    "application/x-apkm",
    "application/x-apks",
    "application/x-xapk",
    "application/xapk",
    "application/vnd.android.xapk",
    "application/vnd.android.apkm",
    "application/apkm",
    "application/vnd.android.apks",
    "application/apks",
    BIN_MIMETYPE
)
const val JSON_MIMETYPE = "application/json"
const val BIN_MIMETYPE = "application/octet-stream"

// Code adapted from Morphe, see third-party/NOTICE for more information
// https://github.com/MorpheApp/morphe-manager/pull/706
const val IMAGE_MIMETYPE = "image/*"
const val WILDCARD_MIMETYPE = "*/*"
